controllers/rollingupgrade_controller.go

```diff
@@ -212,13 +213,20 @@ func (r *RollingUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		IgnoreDrainFailures: r.IgnoreDrainFailures,
 		ReplacementNodesMap: r.ReplacementNodesMap,
 		MaxReplacementNodes: r.MaxReplacementNodes,
+		EarlyCordonNodes:    r.EarlyCordonNodes,
 	}
 
 	// process node rotation
 	if err := rollupCtx.RotateNodes(); err != nil {
 		rollingUpgrade.SetCurrentStatus(v1alpha1.StatusError)
 		rollingUpgrade.SetLabel(v1alpha1.LabelKeyRollingUpgradeCurrentStatus, v1alpha1.StatusError)
 		common.SetMetricRollupFailed(rollingUpgrade.Name)
+
+		// try to uncordon all the cordoned nodes.
+		if _, err2 := rollupCtx.CordonUncordonAllNodes(false); err2 != nil {
```

Comment:
trying to understand why we do Uncordon here? If all old nodes remain in cordon state and there are pods pending, cluster autoscaler will launch new nodes, which is good.

Reply:
No. Think about cases where new nodes aren't coming up because of a faulty AMI. In such cases, we will have to uncordon the nodes or risk pod scaling / failure of new rollouts.

Reply:
Thanks!!!